hasher-matcher-actioner/hmalib/lambdas/api/api_root.py

```diff
@@ -216,3 +273,93 @@ def gen_hash(content_id: str) -> t.Optional[HashResult]:
         "content_hash": record.content_hash,
         "updated_at": record.updated_at.isoformat(),
     }
+
+
+class SignalSourceType(t.TypedDict):
+    type: str
+    count: int
+
+
+class SignalSourceSummary(t.TypedDict):
+    name: str
+    signals: t.List[SignalSourceType]
+    updated_at: str
+
+
+def get_signals() -> t.List[SignalSourceSummary]:
+    """
+    TODO this should be updated to check ThreatExchangeConfig
+    based on what it finds in the config it should then do a s3 select on the files
+    https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Client.select_object_content
+    """
+    mocked_signal_type1a = SignalSourceType(type="HASH_PDQ", count=12456)
+    mocked_signal_type1b = SignalSourceType(type="MOCKED_TYPE1", count=456)
+    mocked_signal_type2a = SignalSourceType(type="HASH_PDQ", count=2456)
+    mocked_signal_type2b = SignalSourceType(type="MOCKED_TYPE2", count=956)
+
+    mocked_signal_source1 = SignalSourceSummary(
+        name="Mocked Signal Source 1",
+        signals=[mocked_signal_type1a, mocked_signal_type1b],
+        updated_at="mocked datetime1",
+    )
+    mocked_signal_source2 = SignalSourceSummary(
+        name="Mocked Signal Source 2",
+        signals=[mocked_signal_type2a, mocked_signal_type2b],
+        updated_at="mocked datetime2",
+    )
+    return [mocked_signal_source1, mocked_signal_source2]
+
+
+class DashboardCount(t.TypedDict):
```

Comment:
Why do we want all of these as TypedDicts and not just classes with the dict keys as class variables?

Reply:
I think that would be a fair amount more code and hard to get field level typing that way. TypedDict is the most minimal solutions I found in a quick search to have some sort of typing on our return data.